### PR TITLE
treewide: Remove usage of isNull

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change

isNull "is deprecated; just write e == null instead" says the [Nix manual](https://nixos.org/nix/manual/#ssec-builtins)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/maintainers/option-usages.nix [hunk 1]
```diff
@@ -145,7 +145,7 @@ let
   displayOptionsGraph =
      let
        checkList =
-         if !(isNull testOption) then [ testOption ]
+         if (testOption != null) then [ testOption ]
```
Comment (jtojnar) [inline nixos/maintainers/option-usages.nix]:
```suggestion
         if testOption != null then [ testOption ]
```

File: nixos/modules/config/sysctl.nix [hunk 2]
```diff
@@ -8,7 +8,7 @@ let
     name = "sysctl option value";
     check = val:
       let
-        checkType = x: isBool x || isString x || isInt x || isNull x;
+        checkType = x: isBool x || isString x || isInt x || (x == null);
```
Comment (jtojnar) [inline nixos/modules/config/sysctl.nix]:
```suggestion
        checkType = x: isBool x || isString x || isInt x || x == null;
```

File: nixos/modules/installer/cd-dvd/iso-image.nix [hunk 3]
```diff
@@ -198,7 +198,7 @@ let
     fi
 
     ${ # When there is a theme configured, use it, otherwise use the background image.
-    if (!isNull config.isoImage.grubTheme) then ''
+    if (config.isoImage.grubTheme != null) then ''
```
Comment (jtojnar) [inline nixos/modules/installer/cd-dvd/iso-image.nix]:
```suggestion
    if config.isoImage.grubTheme != null then ''
```

File: nixos/modules/services/cluster/kubernetes/kubelet.nix [hunk 4]
```diff
@@ -7,9 +7,9 @@ let
   cfg = top.kubelet;
 
   cniConfig =
-    if cfg.cni.config != [] && !(isNull cfg.cni.configDir) then
+    if cfg.cni.config != [] && (cfg.cni.configDir != null) then
```
Comment (jtojnar) [inline nixos/modules/services/cluster/kubernetes/kubelet.nix]:
```suggestion
    if cfg.cni.config != [] && cfg.cni.configDir != null then
```

File: nixos/modules/services/cluster/kubernetes/kubelet.nix [hunk 5]
```diff
@@ -7,9 +7,9 @@ let
...
       throw "Verbatim CNI-config and CNI configDir cannot both be set."
-    else if !(isNull cfg.cni.configDir) then
+    else if (cfg.cni.configDir != null) then
```
Comment (jtojnar) [inline nixos/modules/services/cluster/kubernetes/kubelet.nix]:
```suggestion
    else if cfg.cni.configDir != null then
```

File: nixos/modules/services/continuous-integration/buildkite-agent.nix [hunk 6]
```diff
@@ -236,7 +236,7 @@ in
       };
 
     assertions = [
-      { assertion = cfg.hooksPath == hooksDir || all isNull (attrValues cfg.hooks);
+      { assertion = cfg.hooksPath == hooksDir || all (attrValues cfg.hooks == null);
```
Comment (jtojnar) [inline nixos/modules/services/continuous-integration/buildkite-agent.nix]:
```suggestion
      { assertion = cfg.hooksPath == hooksDir || all (v: v == null) (attrValues cfg.hooks);
```

File: nixos/modules/services/databases/cassandra.nix [hunk 7]
```diff
@@ -219,18 +219,18 @@ in {
   config = mkIf cfg.enable {
     assertions =
       [ { assertion =
-            ((isNull cfg.listenAddress)
-             || (isNull cfg.listenInterface)
-            ) && !((isNull cfg.listenAddress)
-                   && (isNull cfg.listenInterface)
+            ((cfg.listenAddress == null)
```
Comment (jtojnar) [inline nixos/modules/services/databases/cassandra.nix]:
```suggestion
            (cfg.listenAddress == null
```

File: nixos/modules/services/databases/cassandra.nix [hunk 8]
```diff
@@ -219,18 +219,18 @@ in {
...
+             || (cfg.listenInterface == null)
```
Comment (jtojnar) [inline nixos/modules/services/databases/cassandra.nix]:
```suggestion
             || cfg.listenInterface == null
```

**Review by jtojnar — CHANGES_REQUESTED** [review]

File: nixos/modules/services/continuous-integration/buildkite-agent.nix (hunk 6 above)
Comment (JohnAZoidberg) [reply nixos/modules/services/continuous-integration/buildkite-agent.nix]: Ooops

**Review by Profpatsch — APPROVED** [review]
Went through all changes, looks good.

File: nixos/modules/services/logging/logcheck.nix [hunk 9]
```diff
@@ -227,7 +227,7 @@ in
     '';
 
     services.cron.systemCronJobs =
-        let withTime = name: {timeArgs, ...}: ! (builtins.isNull timeArgs);
+        let withTime = name: {timeArgs, ...}: (timeArgs != null);
```
Comment (jtojnar) [inline nixos/modules/services/logging/logcheck.nix]:
```suggestion
        let withTime = name: {timeArgs, ...}: timeArgs != null;
```

File: nixos/modules/services/databases/cockroachdb.nix [hunk 10]
```diff
@@ -164,7 +164,7 @@ in
 
   config = mkIf config.services.cockroachdb.enable {
     assertions = [
-      { assertion = !cfg.insecure -> !(isNull cfg.certsDir);
+      { assertion = !cfg.insecure -> (cfg.certsDir != null);
```
Comment (jtojnar) [inline nixos/modules/services/databases/cockroachdb.nix]:
```suggestion
      { assertion = !cfg.insecure -> cfg.certsDir != null;
```

File: nixos/modules/services/mail/dovecot.nix [hunk 11]
```diff
@@ -384,8 +384,8 @@ in
       { assertion = intersectLists cfg.protocols [ "pop3" "imap" ] != [];
         message = "dovecot needs at least one of the IMAP or POP3 listeners enabled";
       }
-      { assertion = isNull cfg.sslServerCert == isNull cfg.sslServerKey
-          && (!(isNull cfg.sslCACert) -> !(isNull cfg.sslServerCert || isNull cfg.sslServerKey));
+      { assertion = (cfg.sslServerCert == null ) == (cfg.sslServerKey == null )
+          && ((cfg.sslCACert != null) -> !((cfg.sslServerCert == null) || (cfg.sslServerKey == null)));
```
Comment (jtojnar) [inline nixos/modules/services/mail/dovecot.nix]:
```suggestion
          && (cfg.sslCACert != null -> !(cfg.sslServerCert == null || cfg.sslServerKey == null));
```

**Review by jtojnar — COMMENTED** [review]
I would remove the parens, unless necessary for clarity.

**Pull request merged**